app/apphandlers/embeddedAppHandler.go

```diff
@@ -139,13 +139,21 @@ func (a *embeddedApp) StartRudderCore(ctx context.Context, options *app.Options)
 		FeaturesRetryMaxAttempts: 10,
 	})
 
+	dbPool, err := misc.GetDatabaseConnectionPool(ctx, config, statsFactory, "jobsdb")
+	if err != nil {
+		return err
+	}
+	//	Not deferring close here
+	// defer dbPool.Close()
```

Comment:
What the reason for not closing here? Shall we inject dbPool to embeddedAppHandler. So we always use on connection pool

Reply:
just to keep the existing behaviour. We don't close gateway db pool, but the others are closed right away. Since we're sharing the pool now, didn't want to close it.
Just left this comment to find out why the gateway pool wasn't being closed earlier because it's closed in gateway app handler anyway.

Reply:
With the way things are, injecting a pool into `app`s would be a good idea. This would effectively enforce the same database for all components. But I think it'd be nice to have the option of passing different connections pools to different components.

Reply:
what is the existing behaviour you are referring to? I see that every component was closing itself, why have we removed that?

Reply:
we close all `*sql.DB` except gateway db. gateway jobsdb is only `Stop`ped.

Reply:
in any case, closing it everywhere
looks like there was no reason after all